src/DataStructures/Tensor/Expressions/SquareRoot.hpp

```diff
@@ -0,0 +1,88 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+
+#include "DataStructures/Tensor/Expressions/TensorExpression.hpp"
+#include "DataStructures/Tensor/Structure.hpp"
+#include "Utilities/ForceInline.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace TensorExpressions {
+/// \ingroup TensorExpressionsGroup
+/// \brief Defines the tensor expression representing the square root of a
+/// tensor expression that evaluates to a rank 0 tensor
+///
+/// \tparam T the type of the tensor expression of which to take the square
+/// root
+template <typename T>
+struct SquareRoot
+    : public TensorExpression<SquareRoot<T>, typename T::type, tmpl::list<>,
+                              tmpl::list<>, tmpl::list<>> {
+  static_assert(
+      std::is_base_of<TensorExpression<T, typename T::type, tmpl::list<>,
+                                       tmpl::list<>, tmpl::list<>>,
+                      T>::value,
+      "Can only take the square root of a tensor expression that evaluates to "
+      "a rank 0 tensor.");
+  using type = typename T::type;
+  using symmetry = tmpl::list<>;
+  using index_list = tmpl::list<>;
+  using args_list = tmpl::list<>;
+  using structure = Tensor_detail::Structure<symmetry>;
+  static constexpr auto num_tensor_indices = 0;
+
+  SquareRoot(T t) : t_(std::move(t)) {}
+  ~SquareRoot() override = default;
+
+  /// \brief Returns the square root of the component of the tensor evaluated
+  /// from the RHS tensor expression
+  ///
+  /// \details
+  /// SquareRoot only supports tensor expressions that evaluate to a rank 0
+  /// Tensor. This is why, unlike other derived TensorExpression types, there is
+  /// no second variadic template parameter for the generic indices. In
+  /// addition, this is why this template is only instantiated for the case
+  /// where `Structure` is equal to the Structure of a rank 0 Tensor.
+  ///
+  /// \tparam Structure the Structure of the rank 0 Tensor represented by this
+  /// expression
+  /// \param storage_index the storage index of the component of which to take
+  /// the square root
+  /// \return the square root of the component of the tensor evaluated from the
+  /// RHS tensor expression
+  template <typename Structure>
+  SPECTRE_ALWAYS_INLINE decltype(auto) get(
+      const size_t storage_index) const noexcept {
+    static_assert(std::is_same_v<Structure, structure>,
```

Comment:
would it be useful to `ASSERT(storange_index == 0`?

Reply:
Yah, I thought about doing this with `NumberAsExpression`, as well, but I think it's more necessary here since `t_` is a `TensorExpression` we still need to recurse through, and not just a `double`. I just added and squashed in a fixup for this.